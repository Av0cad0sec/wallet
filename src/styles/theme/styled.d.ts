import 'styled-components'
import { ThemeType } from 'grommet'

/* This is the suggested way of declaring theme types */
declare module 'styled-components' {
  export interface DefaultTheme extends ThemeType {}
}
