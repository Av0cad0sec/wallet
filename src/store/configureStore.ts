import { routerMiddleware } from 'connected-react-router'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { createInjectorsEnhancer } from 'redux-injectors'
import createSagaMiddleware from 'redux-saga'

import { createReducer, history } from './reducers'
import { RootState } from 'types'
import { fatalErrorActions } from 'app/state/fatalerror'

export function configureAppStore(state?: Partial<RootState>) {
  const sagaMiddleware = createSagaMiddleware({
    onError: (error, info) => {
      store.dispatch(
        fatalErrorActions.setError({
          message: error.message,
          stack: error.stack,
          sagaStack: info.sagaStack,
        }),
      )
    },
  })
  const { run: runSaga } = sagaMiddleware

  // Create the store with saga middleware
  const middlewares = [sagaMiddleware, routerMiddleware(history)]

  const enhancers = [
    createInjectorsEnhancer({
      createReducer,
      runSaga,
    }),
  ]

  const store = configureStore({
    reducer: createReducer(),
    middleware: [...getDefaultMiddleware(), ...middlewares],
    devTools:
      /* istanbul ignore next line */
      process.env.NODE_ENV !== 'production' || process.env.PUBLIC_URL.length > 0,
    enhancers,
    preloadedState: state,
  })

  return store
}
