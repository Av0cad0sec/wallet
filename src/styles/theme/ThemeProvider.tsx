import { Grommet, ThemeType } from 'grommet'
import * as React from 'react'

import { useThemeSlice } from './slice'

const defaultGrommetTheme: ThemeType = {
  global: {
    font: {
      family: 'Rubik',
    },
  },
}
export const ThemeProvider = (props: { children: React.ReactChild }) => {
  useThemeSlice()

  // const theme = useSelector(selectTheme);
  return (
    // <OriginalThemeProvider theme={theme}>
    <Grommet full theme={defaultGrommetTheme}>
      {React.Children.only(props.children)}
    </Grommet>
    // </OriginalThemeProvider>
  )
}
