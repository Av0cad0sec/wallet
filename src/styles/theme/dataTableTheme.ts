import { IDataTableStyles, ITheme } from 'react-data-table-component'

export const dataTableTheme = {
  background: { default: 'false' },
  highlightOnHover: {
    default: '#88888833',
    text: 'false',
  },
  text: {
    primary: 'false',
    secondary: 'false',
    disabled: 'false',
  },
  sortFocus: {
    default: 'false',
  },
  divider: { default: '#AAAAAAaa' },
} as Partial<ITheme>

export const dataTableCustomStyles: IDataTableStyles = {
  headCells: {
    style: {
      fontSize: '18px',
    },
  },
  rows: {
    style: {
      fontSize: '16x',
    },
  },
}
