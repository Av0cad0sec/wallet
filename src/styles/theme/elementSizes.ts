export const sidebarSmallSizeLogo = 30
export const sidebarMediumSizeLogo = 40
export const mobileToolbarZIndex = 3
export const mobileToolbarButtonsZIndex = mobileToolbarZIndex + 1
export const buildPreviewBannerZIndex = mobileToolbarZIndex - 1

export const mobileFooterNavigationHeight = '4rem'
