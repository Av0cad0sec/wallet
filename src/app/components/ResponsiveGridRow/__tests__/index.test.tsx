import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'
import { ThemeProvider } from 'styles/theme/ThemeProvider'
import { ResponsiveGridRow, PreviewRow, AddressPreviewRow } from '..'

describe('<ResponsiveGridRow />', () => {
  it('should render component', () => {
    const { container } = render(<ResponsiveGridRow label="Type" value="transfer" />)

    expect(container).toMatchSnapshot()
  })
})

const address = 'oasis1qq2vzcvxn0js5unsch5me2xz4kr43vcasv0d5eq4'
const renderComponent = (store: any, address: string) =>
  render(
    <Provider store={store}>
      <ThemeProvider>
        <AddressPreviewRow label="To" value={address} />
      </ThemeProvider>
    </Provider>,
  )

describe('<PreviewRow />', () => {
  it('should render component', () => {
    const { container } = render(<PreviewRow label="Type" value="addEscrow" />)

    expect(container).toMatchSnapshot()
  })
})

describe('<AddressPreviewRow  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore({
      contacts: {
        [address]: {
          address: address,
          name: 'My Contact',
        },
      },
    })
  })

  it('should match snapshot', () => {
    const { container } = renderComponent(store, address)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render without a name', () => {
    renderComponent(store, 'oasis1qqurxkgavtcjjytneumeclx59ds3avjaqg7ftqph')
    expect(screen.queryByText('My Contact')).not.toBeInTheDocument()
  })
})
