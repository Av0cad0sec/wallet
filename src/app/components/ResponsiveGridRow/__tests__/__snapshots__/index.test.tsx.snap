// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResponsiveGridAddressRow  /> should match snapshot 1`] = `
.c0 {
  font-family: Rubik;
  font-size: 18px;
  line-height: 24px;
  background-color: #ffffff;
  color: #16213E;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
}

<div
  class="c0"
  style="min-height: 100dvh;"
>
  <div
    class="c1"
  >
    <span
      class="c2"
    >
      To
    </span>
  </div>
  <div
    class="c3"
  >
    <span
      class="c2"
    >
      My Contact
    </span>
  </div>
  <div
    class="c1"
  >
    <span
      class="c2"
    />
  </div>
  <div
    class="c3"
  >
    <span
      class="c4"
      style="font-family: Roboto mono; letter-spacing: 0;"
    >
      <span
        class="notranslate"
        translate="no"
      >
        oasis1 qq2v zcvx n0js 5uns ch5m e2xz 4kr4 3vca sv0d 5eq4
      </span>
    </span>
  </div>
</div>
`;

exports[`<ResponsiveGridRow /> should render component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1"
    >
      Type
    </span>
  </div>
  <div
    class="c2"
  >
    transfer
  </div>
</div>
`;
